src/react-sortable-tree.js

```diff
@@ -684,40 +695,43 @@ class ReactSortableTree extends Component {
       containerStyle = { height: '100%', ...containerStyle };
 
       const ScrollZoneVirtualList = this.scrollZoneVirtualList;
-      // Render list with react-virtualized
+
+      // Render list with react-window
       list = (
         <AutoSizer>
           {({ height, width }) => (
             <ScrollZoneVirtualList
-              {...scrollToInfo}
+              // ref={this.variableSizeListRef}
+              // to be fixed in the new version of react-dnd-scrollzone
+              direction={rowDirection}
               dragDropManager={dragDropManager}
               verticalStrength={this.vStrength}
               horizontalStrength={this.hStrength}
               speed={30}
-              scrollToAlignment="start"
               className="rst__virtualScrollOverride"
               width={width}
-              onScroll={({ scrollTop }) => {
-                this.scrollTop = scrollTop;
+              onScroll={({ scrollOffset }) => {
+                this.scrollTop = scrollOffset;
               }}
               height={height}
               style={innerStyle}
-              rowCount={rows.length}
-              estimatedRowSize={
+              itemCount={rows.length}
+              estimatedItemSize={
                 typeof rowHeight !== 'function' ? rowHeight : undefined
               }
-              rowHeight={
-                typeof rowHeight !== 'function'
-                  ? rowHeight
-                  : ({ index }) =>
-                      rowHeight({
-                        index,
-                        treeIndex: index,
-                        node: rows[index].node,
-                        path: rows[index].path,
-                      })
+              itemSize={({ index }) =>
+                typeof rowHeight === 'function'
+                  ? rowHeight({
+                      index,
+                      treeIndex: index,
+                      node: rows[index].node,
+                      path: rows[index].path,
+                    })
+                  : rowHeight
               }
-              rowRenderer={({ index, style: rowStyle }) =>
+              {...reactVirtualizedListProps}
+            >
+              {({ index, style: rowStyle }) =>
```

Comment:
This is the issue. It creates new function every render and so every row is mounted and unmounted. You should extract it to extra function and give here only reference to that function so it remains always same.

E.g.
```
this._render = this._render .bind(this); // in constructor

...

_render({ index, style: rowStyle }) {
    const {
      getNodeKey,
    } = mergeTheme(this.props);
    const {
      searchMatches,
      draggedNode,
      draggedDepth,
      draggedMinimumTreeIndex,
      instanceProps,
    } = this.state;

    const treeData = this.state.draggingTreeData || instanceProps.treeData;

    let rows;
    let swapFrom = null;
    let swapLength = null;
    if (draggedNode && draggedMinimumTreeIndex !== null) {
      const addedResult = memoizedInsertNode({
        treeData,
        newNode: draggedNode,
        depth: draggedDepth,
        minimumTreeIndex: draggedMinimumTreeIndex,
        expandParent: true,
        getNodeKey,
      });

      const swapTo = draggedMinimumTreeIndex;
      swapFrom = addedResult.treeIndex;
      swapLength = 1 + memoizedGetDescendantCount({ node: draggedNode });
      rows = slideRows(
        this.getRows(addedResult.treeData),
        swapFrom,
        swapTo,
        swapLength
      );
    } else {
      rows = this.getRows(treeData);
    }
    const matchKeys = {};
    searchMatches.forEach(({ path }, i) => {
      matchKeys[path[path.length - 1]] = i;
    });
    return this.renderRow(rows[index], {
      listIndex: index,
      style: rowStyle,
      getPrevRow: () => rows[index - 1] || null,
      matchKeys,
      swapFrom,
      swapDepth: draggedDepth,
      swapLength,
    })
  }

...

{this._render} // here
```

Should be good to refactor this so there is not much extra computations...

Reply:
But I would also consider it bug in react-dnd, but it is a question if it gets addressed anytime soon ...
https://github.com/react-dnd/react-dnd/search?q=Expected+to+find+a+valid+target&type=Issues